runtime/cudaq/algorithms/gradients/forward_difference.h

```diff
@@ -0,0 +1,54 @@
+/*************************************************************** -*- C++ -*- ***
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ *******************************************************************************/
+
+#pragma once
+
+#include "cudaq/algorithms/gradient.h"
+
+namespace cudaq::gradients {
+
+class forward_difference : public gradient {
```

Comment:
Might want to consider adding some doxygen documentation to this class to describe briefly what it is, etc.